hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/ReadableAccountStore.java

```diff
@@ -142,7 +144,7 @@ private Account getContractLeaf(@NonNull final ContractID id) {
                         // If we didn't find an alias, we will want to auto-create this account. But
                         // we don't want to auto-create an account if there is already another
                         // account in the system with the same EVM address that we would have auto-created.
-                        if (evmAddress.length() > EVM_ADDRESS_LEN && entityNum == null) {
+                        if (!isOfEvmAddressSize(evmAddress) && entityNum == null) {
```

Comment:
I will answer here for both comments. 
You got a point, I have added one more method. 
However, if we have address/key who is < 20, we would fail before that. 
Fixed and comments resolved.